### PR TITLE
Dynamic ref squeeze and unsqueeze

### PR DESCRIPTION
**Pull request by CharlieL7**
* Extends `unsqueeze` and `squeeze` to work for dynamic input shapes
  * Does not handle the `steps` parameter
* Adds some additional negative axes shape tests

### CODE REVIEW

File: test/ref_ops_test.cpp [hunk 1]
```diff
@@ -7361,6 +7361,25 @@ TEST_CASE(unsqueeze_test)
...
+
+    migraphx::shape s1{migraphx::shape::float_type, {{1, 4, 0}, {3, 3, 0}, {3, 3, 0}}};
+    auto p0 = mm->add_parameter("x", s1);
+    mm->add_instruction(migraphx::make_op("unsqueeze", {{"axes", {1}}}), p0);
+    p.compile(migraphx::ref::target{});
+
+    std::vector<float> input_data(4 * 3 * 3);
+    migraphx::parameter_map params0;
+    migraphx::shape input_fixed_shape0{migraphx::shape::float_type, {4, 3, 3}};
+    params0["x"] = migraphx::argument(input_fixed_shape0, input_data.data());
```
Comment (umangyadav) [inline test/ref_ops_test.cpp:7377]:
```suggestion
    params["x"] = migraphx::argument(input_fixed_shape0, input_data.data());
```
You can remove suffix `0` everywhere. makes me wonder if there are multiple params.

Comment (CharlieL7) [reply test/ref_ops_test.cpp:7377]: I'll do this in a separate PR. There are a couple of tests that use multiple params.

**Review by umangyadav — APPROVED** [review]

**Comment by CharlieL7** [comment]
Need dyn_contiguous for the ONNX test. Waiting on that to get merged.

File: src/include/migraphx/op/squeeze.hpp [hunk 2]
```diff
@@ -54,52 +55,90 @@ struct squeeze
...
-                if(old_lens[i] != 1)
+                for(auto i : range(input_shape.ndim()))
                 {
-                    new_lens.push_back(old_lens[i]);
-                    new_strides.push_back(old_strides[i]);
+                    auto dd = input_shape.dyn_dims()[i];
+                    if(dd != one_dyn_dim)
+                    {
+                        dyn_dims.push_back(dd);
+                    }
```
Comment (pfultz2) [inline src/include/migraphx/op/squeeze.hpp:79]: This looks like `std::copy_if`.

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**